### PR TITLE
tailcfg: add caps and SSH action for SSH hauling

### PR DESCRIPTION
**Pull request by skriptble**
Add capabilities and an SSH action for SSH session hauling.

Signed-off-by: Kris Brandow <kris@tailscale.com>

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -1725,6 +1725,7 @@ const (
 	CapabilitySSHRuleIn          = "https://tailscale.com/cap/ssh-rule-in"           // some SSH rule reach this node
 	CapabilityDataPlaneAuditLogs = "https://tailscale.com/cap/data-plane-audit-logs" // feature enabled
 	CapabilityDebug              = "https://tailscale.com/cap/debug"                 // exposes debug endpoints over the PeerAPI
+	CapabilitySSHAggregator      = "https://tailscale.com/cap/ssh-aggregator"        // enable SSH session aggregation
```
Comment (maisem) [inline tailcfg/tailcfg.go:1728]: I think this should be `ssh-aggregator` based on https://tailscale.slack.com/archives/CQQ1AUCCE/p1674669132275479?thread_ts=1674662145.743249&cid=CQQ1AUCCE

cc @bradfitz

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -1940,6 +1943,10 @@ type SSHAction struct {
 	// AllowLocalPortForwarding, if true, allows accepted connections
 	// to use local port forwarding if requested.
 	AllowLocalPortForwarding bool `json:"allowLocalPortForwarding,omitempty"`
+
+	// SessionHauling, if non-empty, is the IP address of a peer to stream this
+	// SSH session's logs to.
+	SessionHauling string `json:"sessionHauling,omitempty"`
```
Comment (maisem) [inline tailcfg/tailcfg.go:1949]: should this be the peers NodeID instead of the IP?
If it has to be an IP then it should likely be `netip.Addr`

Comment (bradfitz) [inline tailcfg/tailcfg.go:1949]: this name sounds like a bool. How about `SessionHaulDest`?

Also, I think we should use a node stable ID here instead of IP address, lest the source not have that address family available. See `func peerAPIBase`.

Comment (bradfitz) [reply tailcfg/tailcfg.go:1949]: Or `SessionHaulTargetNode`.

File: tailcfg/tailcfg.go [hunk 3]
```diff
@@ -1747,6 +1748,8 @@ const (
 	CapabilityWakeOnLAN = "https://tailscale.com/cap/wake-on-lan"
 	// CapabilityIngress grants the ability for a peer to send ingress traffic.
 	CapabilityIngress = "https://tailscale.com/cap/ingress"
+	// CapabilitySSHSessionHaul grants the ability to send SSH session logs.
+	CapabilitySSHSessionHaul = "https://tailscale.com/cap/ssh-session-haul"
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1752]: why is this a cap at all, as opposed to just its presence in the SSHConfig? If control can send one it can send both, but why two?

Comment (bradfitz) [reply tailcfg/tailcfg.go:1752]: oh, this is in the inter-node section! carry on. Maybe reword the comment to make that more clear. I see some of them are more clear about that but then the last few got shorter.

Comment (skriptble) [reply tailcfg/tailcfg.go:1752]: This is checked on the aggregator to know if the SSH server can send logs to it.

File: tailcfg/tailcfg.go (hunk 2 above)
Comment (skriptble) [reply tailcfg/tailcfg.go:1949]: @twitchyliquid64 pointed out that we might want to run the aggregator standalone in the future (maybe behind a subnet router), so being able to select a specific IP:Port that isn't a Tailscale node could be useful for that.

But we could also just change this setting (or add a new one) in the future to handle that, so I'm fine with changing both the name and the type here.

Comment (bradfitz) [reply tailcfg/tailcfg.go:1949]: We'd need a new field to handle a `:port` later anyway, so let's add a new field later if/when we do something else.

**Comment by bradfitz** [comment]
Open a bug too (I don't think we have one right now) and reference it in the commit message (and future commits)

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (skriptble) [reply tailcfg/tailcfg.go:1728]: I'll also rename it to `NodeAttrSSHAggregator` since we've started prefixing them with `NodeAttr` instead of `Capability`.

File: tailcfg/tailcfg.go (hunk 2 above)
Comment (skriptble) [reply tailcfg/tailcfg.go]: Updated to use a `tailcfg.StableNodeID` and updated name.

**Review by bradfitz — APPROVED** [review]
